### PR TITLE
Fix: custom errors

### PR DESCRIPTION
**Pull request by MLopezJ**
Simplify the interface of custom errors

### CODE REVIEW

File: src/utils/getBat.ts [hunk 1]
```diff
@@ -40,7 +39,6 @@ export const getBat = (
 	if ('errors' in maybeValidBat) {
 		return {
 			error: new TypeError({
-				name: 'type error',
 				message: 'error validating type',
```
Comment (coderbyheart) [inline src/utils/getBat.ts]: This is also the same for all TypeErrors, so move it to the class.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: src/converter.ts [hunk 2]
```diff
@@ -95,19 +84,14 @@ export class Warning extends Error {
 	description: string
 
 	constructor({
```
Comment (MLopezJ) [inline src/converter.ts:101]: Give explicit name to `warning` class

File: src/converter.ts [hunk 3]
```diff
@@ -95,19 +84,14 @@ export class Warning extends Error {
...
-		name,
-		message,
-		description,
+		nRFAssetTracker,
```
Comment (MLopezJ) [inline src/converter.ts]: nRFAssetTracker is a name of proyect.

Refer to TypeScript type there as the type. Use `keyof`

File: src/converter.ts [hunk 4]
```diff
@@ -95,19 +84,14 @@ export class Warning extends Error {
...
+		LwM2M,
```
Comment (MLopezJ) [inline src/converter.ts]: LwM2M is the name of the protocol. Update variable name.

File: src/converter.ts [hunk 5]
```diff
@@ -95,19 +84,14 @@ export class Warning extends Error {
...
-		description: string
+		nRFAssetTracker: string
+		LwM2M: string
 	}) {
-		super()
-
-		this.name = name
-		this.message = message
-		this.description = description
+		super(`${nRFAssetTracker} object can not be created`)
```
Comment (MLopezJ) [inline src/converter.ts]: Combine message with description

**Review by coderbyheart — APPROVED** [review]

**Pull request merged**